components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -650,22 +651,28 @@ public static void validateAppliedPolicy(OperationPolicy appliedPolicy,
 
         if (policySpec == null && apiUUID != null) {
             // if policy is not found in the project, policy can be referenced from an existing policy.
-            OperationPolicyData policyData =
-                    provider.getAPISpecificOperationPolicyByPolicyName(appliedPolicy.getPolicyName(),
-                            appliedPolicy.getPolicyVersion(), apiUUID, null, tenantDomain, false);
-            if (policyData != null) {
-                policySpec = policyData.getSpecification();
-                appliedPolicy.setPolicyId(policyData.getPolicyId());
+            if (policyType == null || ImportExportConstants.POLICY_TYPE_API.equalsIgnoreCase(policyType)) {
```

Comment:
Why do we check `null` or `api` type here? Also, don't we have to consider the else case here?